### DISCUSSION TITLE
updating table automatically

### DISCUSSION THREAD

**Discussion by bq-wrongway** [fyne-io/fyne#3550] [Q&A] [answered]
Hi all,
I followed this https://developer.fyne.io/widget/table
And working with table is mostly easy and very nice, however table does not have newtablewithdata variant, so we cant use bindings like with list, what would be best way to rerender table when underlying list changes?
After dataList got new item,table does not render that item,without rebuilding table completely

**Comment by andydotxyz** [comment, marked as answer]
For now call `List.Refresh()` when your data changes.

**Reply by bq-wrongway** [reply]
Well this does not work,
i have 
```
data := make([][]string, 0)

func tableBuilder(list [][]string) fyne.Widget {

	table := widget.NewTable(
		func() (int, int) {
			return len(list), len(list[0])
		},
		func() fyne.CanvasObject {
			return widget.NewLabel("wide content")
		},
		func(i widget.TableCellID, o fyne.CanvasObject) {
			o.(*widget.Label).SetText(list[i.Row][i.Col])

		})

	return table
}
```
before i show my table, i populate the data list, and later when i add another item to the data i call table.Refresh() but nothing happens.

Also as i understand table will accept only string [][] , its not possible  to pass for example **list [] structData**

**Reply by andydotxyz** [reply]
It does work - but if you pass in a `[][]string` to your function it is making a copy of the data - so when you change it back at the source the new data is not known to the table. You'll probably want to pass a pointer to that data, or pass it as part of a wider "app struct"

**Reply by bq-wrongway** [reply]
You are right, thank you very much.
